File: AlemayehuKettema/document/client/python/supplier_quatation.py
@frappe.whitelist()
def get_supplier_quotations(request_for_quotation):
    # Fetch Supplier Quotations matching the given RFQ
    supplier_quotations = frappe.get_list('Supplier Quotation', 
                                          filters={'quotation_number': request_for_quotation}, 
                                          fields=['name', 'supplier'])
    
    if not supplier_quotations:
        frappe.throw("No supplier quotations found for the selected request for quotation.")

    items_to_add = []

    for quotation in supplier_quotations:
        # Fetch items for each Supplier Quotation
        items = frappe.get_list('Supplier Quotation Item', 
                                filters={'parent': quotation.name}, 
                                fields=['item_code', 'item_name', 'uom', 'qty', 'rate', 'amount'])

        for item in items:
            items_to_add.append({
                'supplier': quotation.supplier,
                'item_code': item.item_code,
                'item_name': item.item_name,
                'uom': item.uom,
                'quantity': item.qty,
                'rate': item.rate,
                'amount': item.amount
            })
    
    # Return the items to be added to the 'Supplier Quotation Comparison Table 2'
    return items_to_add

File: AlemayehuKettema/document/client/js/supplier_quatation.js
frappe.ui.form.on('Supplier Quotation', {
    quotation_numbers: function (frm,cdt,cdn) {
            frappe.model.with_doc('Request for Quotation', frm.doc.quotation_numbers, function () {
                let source_doc = frappe.model.get_doc('Request for Quotation',frm.doc.quotation_numbers);
                $.each(source_doc.items, function (index, source_row) {
			            const target_row = frm.add_child('items');
                    	target_row.item_code = source_row.item_code;
                        target_row.uom = source_row.uom;
                        target_row.description = source_row.description;
                        target_row.qty = source_row.qty;
                        target_row.warehouse = source_row.warehouse;
                });
                frm.refresh_field('items');
            });
    },
    pr_number: function (frm,cdt,cdn) {
        frappe.model.with_doc('Purchase Requisition', frm.doc.pr_number, function () {
            let source_doc = frappe.model.get_doc('Purchase Requisition',frm.doc.pr_number);
            $.each(source_doc.items, function (index, source_row) {
                    const target_row = frm.add_child('items');
                    target_row.item_code = source_row.item_code;
                    target_row.uom = source_row.uom;
                    target_row.description = source_row.description;
                    target_row.qty = source_row.qty;
                    target_row.warehouse = source_row.warehouse;
            });
            frm.refresh_field('items');
        });
},
});